src/org/labkey/test/util/search/SearchAdminAPIHelper.java

```diff
@@ -28,26 +29,54 @@
 import java.util.Map;
 
 import static org.junit.Assert.assertEquals;
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_PAGE;
 
 public abstract class SearchAdminAPIHelper
 {
+
+    public static void waitForIndexer()
+    {
+        waitForIndexer(WAIT_FOR_PAGE);
+    }
+
     /**
      * Wait for search indexer to be idle via SearchController.WaitForIndexerAction
      */
     @LogMethod(quiet = true)
-    public static void waitForIndexer()
+    public static void waitForIndexer(int timeout)
     {
         // Invoke a special server action that waits until all previous indexer tasks are complete
-        int response = WebTestHelper.getHttpResponse(WebTestHelper.buildURL("search", "waitForIndexer")).getResponseCode();
-        assertEquals("WaitForIndexer action timed out", HttpStatus.SC_OK, response);
+        var cmd = new PostCommand("search", "waitForIndexer");
+        cmd.setTimeout(timeout);
+        executeWaitForIndexer(cmd);
     }
 
-    @LogMethod(quiet = true)
     public static void waitForIndexerBackground()
+    {
+        waitForIndexerBackground(WAIT_FOR_PAGE);
+    }
+
+    @LogMethod(quiet = true)
+    public static void waitForIndexerBackground(int timeout)
     {
         // Invoke a special server action that waits until all previous indexer tasks are complete, even wait for background indexing tasks to complete (e.g. deleteContainer)
-        int response = WebTestHelper.getHttpResponse(WebTestHelper.buildURL("search", "waitForIndexer", Map.of("priority","background"))).getResponseCode();
-        assertEquals("WaitForIndexer action timed out", HttpStatus.SC_OK, response);
+        var cmd = new PostCommand("search", "waitForIndexer");
+        cmd.setTimeout(timeout);
+        cmd.setParameters(Map.of("priority","background"));
```

Comment:
Commands get crabby sometimes if you pass an immutable Map to `setParameters`. You need to pass in a mutable collection.
```suggestion
        cmd.setParameters(new HashMap<>(Map.of("priority", "background")));
```